### DISCUSSION TITLE
Support for popup windows for OAuth flow

### DISCUSSION THREAD

**Discussion by sazzer** [wailsapp/wails#2229] [Q&A]
Hi there,

I've been using Wails to great effect so far, but I've hit a slight snag. For the application I'm building, I need to be able to authenticate the user against a remote service. As such, I need to be able to open a new browser window to an OAuth URL, have them go through the authentication process, and then redirect back to the application so that I can obtain an access token to call the remote service.

Is this possible? In particular, I'm meaning:
* Is it possible to open a new window for the authentication flow?
* Is it possible to have the redirect from the remote server come back to Wails for it to handle the result?

Cheers

**Comment by annismckenzie** [comment]
I have the same issue. I'll start by investigating how to register a protocol for the application because that's the link between the browser window where the authentication happens and where, after that succeeds, the callback is triggered in that has to "open" the application again.

**Reply by annismckenzie** [reply]
It looks like this is being implemented in #3000 and that will be merged soon. It'll probably still be a while and I don't know whether it'll make it into a 2.x release or only arrive in 3.0 (which doesn't even have its first alpha release yet 😅).
